Classes/DDFileLogger.m

```diff
@@ -1531,3 +1544,101 @@ BOOL doesAppRunInBackground() {
 }
 
 #endif
+
+static NSUInteger kMaximumBytesCountInBuffer = (1 << 10) * (1 << 10); // 1 MB.
+static NSUInteger kDefaultBytesCountInBuffer = (1 << 10);
+@interface DDFileLoggerWithBuffer : DDFileLogger
+@property (assign, nonatomic, readwrite) NSUInteger maximumBytesCountInBuffer;
+@end
+
+@interface DDFileLoggerWithBuffer() {
+    NSOutputStream *_bufferStream;
+    NSUInteger _bufferSize;
+}
+@end
+
+@interface DDFileLoggerWithBuffer (StreamManipulation)
+- (void)flushBuffer;
+- (void)dumpBufferToDisk;
+- (void)appendToBufferData:(NSData *)data;
+@end
+
+@implementation DDFileLoggerWithBuffer (StreamManipulation)
+- (void)flushBuffer {
+    // do something.
+    [_bufferStream close];
+    _bufferStream = nil;
+    _bufferSize = 0;
+}
+- (void)dumpBufferToDisk {
+    // do something.
+    __auto_type data = (NSData *)[_bufferStream propertyForKey:NSStreamDataWrittenToMemoryStreamKey];
+    [super logData:data];
+    [self flushBuffer];
+}
+- (void)appendToBufferData:(NSData *)data {
+    __auto_type length = data.length;
+    if (data.length != 0) {
+        if (_bufferStream == nil) {
+            _bufferStream = [[NSOutputStream alloc] initToMemory];
+            [_bufferStream open];
+            _bufferSize = 0;
+        }
+        const uint8_t *appendedData = calloc(length, sizeof(uint8_t));
+        [data getBytes:(void *)appendedData length:length];
+        if (appendedData != NULL) {
+            [_bufferStream write:appendedData maxLength:length];
+        }
+        if (appendedData != NULL) {
+            free((void *)appendedData);
+        }
+        _bufferSize += _bufferSize + length;
+    }
+}
+@end
+
+@implementation DDFileLoggerWithBuffer
+@synthesize maximumBytesCountInBuffer = _maximumBytesCountInBuffer;
+- (instancetype)init {
+    if (self = [super init]) {
+        self.maximumBytesCountInBuffer = kDefaultBytesCountInBuffer;
+    }
+    return self;
+}
+
+#pragma mark - Subclass
+- (void)logData:(NSData *)data {
+    [self appendToBufferData:data];
+}
+
+#pragma mark - Initialization
+- (void)cleanup {
+    [self dumpBufferToDisk];
+    [super cleanup];
+}
+@end
+
+@interface DDFileLoggerWithBuffer (Logging)
+- (BOOL)shouldDumpBufferToDisk;
+- (void)putDataIntoBufferOrToDisk:(NSData *)data;
+@end
+
+@implementation DDFileLoggerWithBuffer (Logging)
+- (BOOL)shouldDumpBufferToDisk {
+    return _bufferSize > _maximumBytesCountInBuffer;
+}
+- (void)putDataIntoBufferOrToDisk:(NSData *)data {
```

Comment:
Missing newline before method

Reply:
@ffried 
Is there any .clang_format file in the repository?
It would be nice to have it to prevent further issues in style.

Reply:
I've started one, but could not get everything to be styled correctly. Also, I'm not a big fan of "hard styling rules". There are a few basics, but e.g. I don't care if the implementation of a method is inline with its declaration as long as it's readable (e.g. `- (BOOL)shouldDoWork { return NO; }`)

My general advice here: Look at the code around you and style accordingly. 😉